mesh/ncmesh.hpp

```diff
@@ -32,17 +32,38 @@ namespace mfem
 /** Represents the index of an element to refine, plus a refinement type.
     The refinement type is needed for anisotropic refinement of quads and hexes.
     Bits 0,1 and 2 of 'ref_type' specify whether the element should be split
-    in the X, Y and Z directions, respectively (Z is ignored for quads). */
+    in the X, Y and Z directions, respectively (Z is ignored for quads). The
+    refinement spacing or scale in each direction is a number in (0,1), with the
+    default 0.5 meaning bisection. This linear scale parameter defines the
+    position of the refinement between the beginning (0) and end (1) of the
+    reference element in each direction. */
 struct Refinement
 {
-   enum : char { X = 1, Y = 2, Z = 4, XY = 3, XZ = 5, YZ = 6, XYZ = 7 };
    int index; ///< Mesh element number
-   char ref_type; ///< refinement XYZ bit mask (7 = full isotropic)
-
+   /** Type is used only for input. The struct stores the refinement type in the
+       triple @a scale, which specifies the refinement scale in each dimension,
+       with 0 representing no refinement in that dimension. */
+   enum : char { X = 1, Y = 2, Z = 4, XY = 3, XZ = 5, YZ = 6, XYZ = 7 };
+   using ScaledType = std::pair<char, real_t>;
+   real_t s[3];  /// Refinement scale in each dimension
    Refinement() = default;
-
-   Refinement(int index, int type = Refinement::XYZ)
-      : index(index), ref_type(type) {}
+   /// Refinement type XYZ, with scale 0.5.
+   Refinement(int index);
+   /// Default case of empty list @a refs is XYZ with scale 0.5.
+   Refinement(int index, const std::initializer_list<ScaledType> &refs);
+   /// Refine element with a single type and scale in all dimensions.
+   Refinement(int index, char type, real_t scale = 0.5);
+   Refinement(int index, int type, real_t scale = 0.5);
```

Comment:
All the `int` arged interfaces shouldn't exist, they're a type conversion issue waiting to happen. If back the enum with char is important enough to do in the first place, then avoiding all the type conversions back and forth should be too.

Reply:
Removed in e5cfc35f87969a015c88f892f8927267bd9ef4c5